day-3/engine/engine.py

```diff
@@ -0,0 +1,64 @@
+import re
+from engine.number import Number
+
+SYMBOL = '#'
+GEAR = '*'
+DIGITS = '0123456789'
+
+class Engine():
+    def __init__(self):
+        self.schematic = []
+        self.rows = 0
+        self.columns = 0
+        self.numbers = []
+        self.symbols = set()
+        self.gears = dict()
+
+    def process_schematic(self, input):
+        self.make_symbols_consistent(input)
+        self.rows = len(self.schematic)
+        self.columns = len(self.schematic[0]) - 1
+
+        self.find_numbers_and_symbols()
```

Comment:
Here is example of place where you could use a static factory method to create your Engine object. Otherwise, you will always need to call `process_schematic` after creating a object - i.e. just the object created by the constructor is not enough/valid to be used.

Reply:
Tried this suggestion, can you let me know if it looks right?